### PR TITLE
 PLENTY-318_use_new_method_setDocuments_for_navigation

### PR DESCRIPTION
**Pull request by dblaichinger**

### CODE REVIEW

File: src/Services/FallbackSearchService.php [hunk 1]
```diff
@@ -59,7 +53,7 @@ public function handleSearchQuery(Request $request, ExternalSearch $externalSear
      * @param ExternalSearch $externalSearch
      * @return array
      */
-    private function getSearchResults(Request $request, ExternalSearch $externalSearch)
+    public function getSearchResults(Request $request, ExternalSearch $externalSearch)
```
Comment (TheKeymaster) [inline src/Services/FallbackSearchService.php:56]: If this is only public for tests, it should also be possible to mock this when you set it to `protected`. This should also be possible for `createResponseFromSearchResult`. I am not 100% certain but I think that these can still be mocked.

File: tests/Services/SearchServiceTest.php [hunk 2]
```diff
@@ -1209,6 +1214,54 @@ public function testInitialNavigationPageGetsResultCountFromPlenty()
...
+        $requestMock = $this->getMockBuilder(HttpRequest::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $externalSearchServiceMock = $this->getMockBuilder(ExternalSearch::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $externalSearchServiceMock->expects($this->once())
+            ->method('setDocuments')
+            ->willThrowException(new Exception('Method setDocuments doesn\t exist'));
```
Comment (TheKeymaster) [inline tests/Services/SearchServiceTest.php:1264]:
```suggestion
            ->willThrowException(new Exception('Method setDocuments doesn\'t exist'));
```

File: tests/Services/SearchServiceTest.php [hunk 3]
```diff
@@ -1209,6 +1214,54 @@ public function testInitialNavigationPageGetsResultCountFromPlenty()
...
+
+        $this->fallbackSearchService->expects($this->once())
+            ->method('createResponseFromSearchResult')
+            ->willReturn($responseMock);
+
+        $requestMock = $this->getMockBuilder(HttpRequest::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $externalSearchServiceMock = $this->getMockBuilder(ExternalSearch::class)
```
Comment (TheKeymaster) [inline tests/Services/SearchServiceTest.php:1259]: So using `stdClass` was not possible, could you explain what was the issue?

**Review by TheKeymaster — CHANGES_REQUESTED** [review]

File: src/Services/FallbackSearchService.php (hunk 1 above)
Comment (dblaichinger) [reply src/Services/FallbackSearchService.php:56]: It's necessary to call those methods in `SearchService::doNavigation` that's why I set it to public.

File: tests/Services/SearchServiceTest.php (hunk 2 above)
Comment (dblaichinger) [reply tests/Services/SearchServiceTest.php:1264]: Wow that is a really nice feature 👍

File: tests/Services/SearchServiceTest.php (hunk 3 above)
Comment (TheKeymaster) [reply tests/Services/SearchServiceTest.php:1259]: Damn, understandable. Resolved.

**Comment by dblaichinger** [comment]
Closed, renamed branch and new PR is #251

**Pull request closed without merging**